kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/security/auth/SimpleAclAuthorizer.java

```diff
@@ -0,0 +1,147 @@
+                                            .auth_policies
+                                            .getTopicAuthentication()
+                                            .get(topicName.toString());
+                                    if (topicRoles != null && role != null) {
+                                        // Topic has custom policy
+                                        Set<AuthAction> topicActions = topicRoles.get(role);
+                                        if (topicActions != null && topicActions.contains(action)) {
+                                            permissionFuture.complete(true);
+                                            return;
+                                        }
+                                    }
+
+                                    Map<String, Set<AuthAction>> namespaceRoles = policies.get().auth_policies
+                                            .getNamespaceAuthentication();
+                                    Set<AuthAction> namespaceActions = namespaceRoles.get(role);
+                                    if (namespaceActions != null && namespaceActions.contains(action)) {
+                                        permissionFuture.complete(true);
+                                        return;
+                                    }
+                                }
+                                permissionFuture.complete(false);
+                            }).exceptionally(ex -> {
+                                log.warn("Client with Principal - {} failed to get permissions for resource - {}. {}",
+                                        principal, topicName, ex.getMessage());
+                                permissionFuture.completeExceptionally(ex);
+                                return null;
+                            });
+                } catch (Exception e) {
+                    log.warn("Client with Principal - {} failed to get permissions for topic - {}. {}",
+                            principal, topicName, e.getMessage());
+                    permissionFuture.completeExceptionally(e);
+                }
+            }
+        });
+
+        return permissionFuture;
+    }
+
+    private CompletableFuture<Boolean> isSuperUser(String role) {
+        Set<String> superUserRoles = getPulsarService().getConfiguration().getSuperUserRoles();
+        return CompletableFuture.completedFuture(role != null && superUserRoles.contains(role));
+    }
+
+    @Override
+    public CompletableFuture<Boolean> canLookupAsync(KafkaPrincipal principal, Resource resource) {
+        CompletableFuture<Boolean> canLookupFuture = new CompletableFuture<>();
+        authorize(principal, AuthAction.produce, resource)
+                .whenComplete((hasProducePermission, ex) -> {
+                    if (ex != null) {
+                        log.warn("Check produce permission error:{}", ex.getMessage());
```

Comment:
I think we should change the log level to debug, because each time a consumer without produce permission connected to KoP, it would always reach here and there could be a lot of warnings.

Reply:
Sorry I'm wrong that it won't reach here because `ex` will be null and `hasProducePermission` will be false. Resolve this conversation.